### ISSUE TITLE
Restart fails

### ISSUE THREAD

**Issue by andig** [gokrazy/mkfs#3]
Coming from https://github.com/gokrazy/gokrazy/issues/154.

Restarting doesn't work when run through `qemu`. This is `stderr`:

```
2022/11/20 18:01:38 gokrazy: attempt 0, starting ["/user/mkfs"]
2022/11/20 18:01:38 Found mountpoint "/"
2022/11/20 18:01:38 Found mountpoint "/dev"
2022/11/20 18:01:38 Found mountpoint "/tmp"
2022/11/20 18:01:38 Found mountpoint "/dev/pts"
2022/11/20 18:01:38 Found mountpoint "/dev/shm"
2022/11/20 18:01:38 Found mountpoint "/run"
2022/11/20 18:01:38 Found mountpoint "/proc"
2022/11/20 18:01:38 Found mountpoint "/sys"
2022/11/20 18:01:38 Found mountpoint "/sys/fs/cgroup"
2022/11/20 18:01:38 No /perm mountpoint found. Creating file system on /dev/vda4
2022/11/20 18:01:38 Writing self-contained mke2fs to /tmp/gokrazy-mkfs-204447501
2022/11/20 18:01:38 [/tmp/gokrazy-mkfs-204447501/mke2fs -t ext4 /dev/vda4]
mke2fs 1.46.5 (30-Dec-2021)
Post "http://localhost/reboot": dial tcp [::1]:80: connect: connection refused
2022/11/20 18:01:43 gokrazy: process should not be supervised, stopping
2022/11/20 18:01:43 gokrazy: exit status 125
```

@stapelberg I guess that's because I have moved the host application to :80 (and gokrazy is on :443). Would you like a PR for a command line option to change the port of `mkfs` or do you see a better approach?

**Comment by stapelberg** [comment]
> @stapelberg I guess that's because I have moved the host application to :80 (and gokrazy is on :443). Would you like a PR for a command line option to change the port of `mkfs` or do you see a better approach?

There is a better approach indeed: like `breakglass`, we should read the parameters for loopback gokrazy API requests from the file system: https://github.com/gokrazy/breakglass/blob/2c1eed342dd4dc5ee4f00bbc1f7bd62fd49171c4/breakglass.go#L118

Feel free to copy this into mkfs and send a PR for now, but we should eventually have a package in `gokrazy/internal` which helps with this.

**Comment by stapelberg** [comment]
Fixed with PR #4
